### PR TITLE
Added formValidation, errorHandler. Refactored addEventListener

### PR DESCRIPTION
**Pull request by demakoff**

### CODE REVIEW

File: formValidation.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    window.formValidation = {
+        validateArgsByEvent: validateArgsByEvent,
+        dataTypesRegExp: dataTypesRegExp
+    };
+
+    function validateArgsByEvent(args, methodName) {
+        let resultPromise = Promise.resolve([]);
+
+        args.forEach((argNode) => {
+            resultPromise = resultPromise.then(validateArg.bind(this, argNode, methodName));
```
Comment (Boooober) [inline formValidation.js]: Consider passing `null` as a context if you do not care about it. In order to make it readable.

File: formValidation.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        });
+        return resultPromise;
```
Comment (Boooober) [inline formValidation.js]: Could we shorten this function like below?

```
function validateArgsByEvent(args, methodName) {
    return Promise.all(args.map(argNode => validateArg(argNode, methodName)));
}
```

File: formValidation.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            const validateRule = dataTypesRegExp[type];
+            if (validateRule && !validateRule.test(value)) {
+                throw {
+                    type: 'argumentValidationError',
+                    argName: name,
+                    methodName
+                };
+            }
+            argsValues.push(value);
+            return argsValues;
```
Comment (Boooober) [inline formValidation.js]: If function above is acceptable, we could resolve valid value or throw error.

File: main.js [hunk 4]
```diff
@@ -561,27 +557,29 @@
...
-                let valid = true;
-                args.forEach((arg) => {
-                    arg.classList.remove('invalid');
-                    const argType = arg.getAttribute('data-arg-type');
-                    const validateRule = dataTypesRegExp[argType];
-                    if (validateRule && !validateRule.test(arg.value)) {
-                        valid = false;
-                        arg.classList.add('invalid');
+            button.addEventListener('click', (event) => {
+                const methodName = event.target.getAttribute('data-method-name');
```
Comment (Boooober) [inline main.js]: Might be better to extract eventHandler function. We should be able to remove event listener in future.

File: formValidation.js (hunk 1 above)
Comment (demakoff) [reply formValidation.js]: ok, agree

File: formValidation.js (hunk 2 above)
Comment (demakoff) [reply formValidation.js]: I believe, we can

File: main.js (hunk 4 above)
Comment (demakoff) [reply main.js]: Don't mind

**Review by demakoff — COMMENTED** [review]
-

**Pull request merged**